frontend/src/metabase-lib/filter.ts

```diff
@@ -100,6 +109,63 @@ export function removeFilters(query: Query, stageIndex: number): Query {
   );
 }
 
+export function appendStageIfSummarized(query: Query): Query {
```

Comment:
This function belongs in `query.ts` but it needs to reference `aggregation.ts` and `breakout.ts` which would cause a circular dependency. We don't really have a file with cross-clause high-level MLv2 utils, so I'm defining it as close to the usage as possible.